integration-tests/api/__tests__/admin/analytics-config.js

```diff
@@ -0,0 +1,215 @@
+const path = require("path")
+const startServerWithEnvironment =
+  require("../../../helpers/start-server-with-environment").default
+const { useApi } = require("../../../helpers/use-api")
+const { useDb } = require("../../../helpers/use-db")
+const {
+  simpleAnalyticsConfigFactory,
+} = require("../../factories/simple-analytics-config-factory")
+const adminSeeder = require("../../helpers/admin-seeder")
+
+const adminReqConfig = {
+  headers: {
+    Authorization: "Bearer test_token",
+  },
+}
+
+jest.setTimeout(30000)
+describe("[MEDUSA_FF_ANALYTICS] /admin/analytics-config", () => {
+  let medusaProcess
+  let dbConnection
+
+  beforeAll(async () => {
+    const cwd = path.resolve(path.join(__dirname, "..", ".."))
+    const [process, connection] = await startServerWithEnvironment({
+      cwd,
+      env: { MEDUSA_FF_ANALYTICS: true },
+    })
+    dbConnection = connection
+    medusaProcess = process
+  })
+
+  afterAll(async () => {
+    const db = useDb()
+    await db.shutdown()
+
+    medusaProcess.kill()
+  })
+
+  describe("GET /admin/analytics-config", () => {
+    beforeEach(async () => {
+      try {
+        await adminSeeder(dbConnection)
+      } catch (e) {
+        console.error(e)
+      }
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      await db.teardown()
+    })
+
+    it("should retrieve config for logged in user if it exists", async () => {
+      // Create config
+      await simpleAnalyticsConfigFactory(dbConnection)
+
+      const api = useApi()
+      const response = await api.get(`/admin/analytics-configs`, adminReqConfig)
+
+      expect(response.data).toMatchSnapshot({
+        analytics_config: {
+          id: expect.any(String),
+          user_id: "admin_user",
+          opt_out: false,
+          anonymize: false,
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+      })
+    })
+
+    it("should return 404 if no config exists", async () => {
+      const api = useApi()
```

Comment:
Add expect assertions since no assertions is a success test
```suggestion
      expect.assertions(2)
      const api = useApi()
```

Reply:
don't think the expect assertions is necessary here. it just means that we expect 3 expects to be called before the tests timeout. It seam a bit redundant no?

Reply:
not neccessary, this comment was made before the test was refactored at which point the assertions was made in the catch block 😄 closing the comment